### PR TITLE
feat: Edit updates

### PR DESCRIPTION
**Pull request by ncomerci**
image
image

### CODE REVIEW

File: src/intl/en.json [hunk 1]
```diff
@@ -1078,6 +1085,8 @@
     "proposal_update": {
       "title": "Publish New Grant Update",
       "description": "Share your grant updates with the Decentraland Community. Use this space to talk about the progress but also to raise issues or blockers you might have with your project. Feel free to add any relevant information or links to demo what you’ve been up to.",
+      "edit_title": "Edit Grant Update",
+      "edit_description": "...",
```
Comment (ncomerci) [inline src/intl/en.json:1089]: Pending description

**Review by 1emu — APPROVED** [review]
Nnnnnnaiss

File: src/pages/edit/update.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+import Update from '../submit/update'
+
+function update() {
```
Comment (andyesp) [inline src/pages/edit/update.tsx:5]: I'd rename this to `UpdateEditPage`

File: src/entities/Updates/routes.ts [hunk 3]
```diff
@@ -117,10 +117,12 @@ async function updateProposalUpdate(req: WithAuth<Request<{ proposal: string }>>
     throw new RequestError(`Update not found: "${id}"`, RequestError.NotFound)
   }
 
-  if (update?.completion_date) {
-    throw new RequestError(`Update already completed: "${update.id}"`, RequestError.BadRequest)
+  if (!update?.completion_date) {
```
Comment (andyesp) [inline src/entities/Updates/routes.ts:120]: This function is being used to update pending proposal updates. Pending updates don't have completion_date.
If I understand correctly, if you make this change, users with tier 3-6 grants can't post updates.

File: src/pages/submit/update.tsx [hunk 4]
```diff
@@ -202,46 +228,54 @@ export default function Update() {
...
     return <LoadingView />
   }
 
-  if (updateId && (updateState.error || update?.status === UpdateStatus.Late || update?.status === UpdateStatus.Done)) {
+  const isDisabled =
+    !isEdit &&
+    updateId &&
+    (updateState.error || update?.status === UpdateStatus.Late || update?.status === UpdateStatus.Done)
+
+  const isUserEnabledToEdit = isEdit && update?.author === account
```
Comment (andyesp) [inline src/pages/submit/update.tsx:242]: Maybe we should let update authors and proposal coauthors edit.

**Review by andyesp — CHANGES_REQUESTED** [review]

File: src/components/Modal/EditUpdateModal/EditUpdateModal.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import { Header } from 'decentraland-ui/dist/components/Header/Header'
+import { Modal, ModalProps } from 'decentraland-ui/dist/components/Modal/Modal'
+
+import '../ProposalModal.css'
+
+type Props = Omit<ModalProps, 'children'> & {
+  onClickAccept?: (e: React.MouseEvent<unknown>) => void
+}
+
+export function EditUpdateModal({ onClickAccept, loading, ...props }: Props) {
```
Comment (andyesp) [inline src/components/Modal/EditUpdateModal/EditUpdateModal.tsx:17]: Is this modal in the design? I think we can navigate the user to the edit page by clicking on the edit button in the update menu without showing this modal...

Comment (andyesp) [reply src/components/Modal/EditUpdateModal/EditUpdateModal.tsx:17]: Sorry, this is like a confirmation modal after saving the edited update right?

Comment (ncomerci) [reply src/components/Modal/EditUpdateModal/EditUpdateModal.tsx:17]: Yes, it is in the requirements!

image

File: src/pages/submit/update.tsx (hunk 4 above)
Comment (ncomerci) [reply src/pages/submit/update.tsx:242]: According to Gino, only the author of the update can edit it

**Review by andyesp — APPROVED** [review]

**Pull request merged**